### ISSUE TITLE
Restore appears to restart every time the activity is opened

### ISSUE THREAD

**Issue by mattfbacon** [stevesoltys/backup#815]
While the Restore is active, if I do something else and then return to the activity, the progress bar starts back at the beginning and I see the same apps being installed one-by-one again.

**Comment by mattfbacon** [comment]
The half-restored state for installing missing apps disappears as well, which is annoying because we are supposed to be using other app stores to install these apps at this stage.

**Comment by grote** [comment]
What version of Seedvault are you using? Can you share a screenshot of the About screen please?

**Comment by grote** [comment]
Do you see a notification for restore while restore is running? After restore has restarted, can you leave the restore flow and export logs for us to analyze? I imagine, you can easily reproduce the "restart" behavior? Helpful would also be a screencast, so we understand exactly what is happening here.

**Comment by mattfbacon** [comment]
Version 14-3.3

I closed the app by pressing the square button (it goes to the home screen, recents has always been super buggy, that's just how it is). Then I go back to seedvault and the progress resets after a bit.

[Video contents described above](https://github.com/user-attachments/assets/c36eb5fb-4cac-4244-b2c5-f6c0f8a0948d)

**Comment by mattfbacon** [comment]
There is also some weirdness because grapheneos embeds Seedvault into settings, i.e., it has no entry in the Apps menu (it's a system app). I wanted to check there if notifications are disabled for Seedvault because I didn't see any progress notification for the Restore.

**Comment by grote** [comment]
Ok version 14-3.3 is super outdated. This version doesn't support restore after setup wizard (where you can leave the app). If Graphene enabled this for this version, it is on them. Calyx didn't support restore after SUW for this Seedvault version. 

Are you sure you are on latest Graphene release?

**Comment by mattfbacon** [comment]
I am on the 14 tag of GrapheneOS which is supposedly the development branch for the Pixel 5a.

**Comment by grote** [comment]
Are you self-building the OS then? Maybe try pull from android14 branch of this repo into `packages/apps/Seedvault`.

**Comment by mattfbacon** [comment]
This is the source: https://github.com/GrapheneOS/platform_external_seedvault/tree/14
Looks like the last common commit is 695c33e.
I don't want to be try directly from this repo because I think some of the GrapheneOS patches are necessary but I could try rebasing their patches on top of this repo.

**Comment by grote** [comment]
Yes try rebasing. The commits touching Android.bp files are probably needed. Could also cherry-pick those in case rebase fails.

**Comment by mattfbacon** [comment]
So I was not able to get the new Seedvault to work with Graphene and I switched back to Calyx anyway so I will close this because it's probably just due to the outdated version.
